packages/core/src/modules/proofs/services/ProofService.ts

```diff
@@ -1137,6 +1143,27 @@ export class ProofService {
       )
     }
   }
+
+  public assertAttributeNamesDoNotMatch(proofRequest: ProofRequest) {
+    const attributes = Array.from(proofRequest.requestedAttributes.values()).map((a) => a.name)
+    const duplicates = attributes.filter((item, index) => attributes.indexOf(item) != index)
```

Comment:
Maybe I don't see it but it looks like you are comparing each items index with its own index.

Couldn't you create a `Set` of the array and compare those lengths?

e.g.

```js
const attributes = Array.from(proofRequest.requestedAttributes.values()).map((a) => a.name)
const uniqueAttributes = new Set(attributes)
// Ignore the bad naming
const areAttributesTrulyUnique = attributes.length === uniqueAttributes.size
```

Reply:
Ah I see now how it is supposed to be. `indexOf` returns the first index and if it occured a second time those indices will not be the same. 

Ignore my comment :).